### PR TITLE
Sprite SVGs

### PR DESCRIPTION
**Pull request by mainej**
- Removes SVGs from the JS
- Re-uses SVGs in static content

But,
- Makes initial document load larger, so still TODO:
  - [x] Serve sprite separately (see comments below)

Also
- [x] Move background SVGs into sprite
- [ ] ~~Move various spinner SVGs to sprite~~

Long term improvements:
- [ ] Unify `svg/micro-dollar-sign` and `svg/large-dollar`, possibly using `vector-effect: non-scaling-stroke`
- [x] Get design permission to use same icons on /help page and help section of footer (text, call, email)
- [ ] ~~Convert `svg`s from `px` to `em`, so they scale with font size~~

### CODE REVIEW

**Comment by sheelc** [comment]
It'd be great if there was a way to serve the sprite separately -- hopefully it's possible. As a heads up the inlined svg is already GZIP since the [whole HTML response is GZIP-ed](https://github.com/Mayvenn/storefront/blob/42c44350f2deab92397df9e449a2aad62c69e61b/src/storefront/jetty.clj)

**Comment by mainej** [comment]
Yeah, as is this change just means moving around the SVG code, not serving it separately. But even so, turning the SVGs back into XML gets us slightly closer to testing HTTP/2.

However... I think I found an article with the trick to [serving the sprite separately but referencing it in background images](https://codepen.io/jennhi/post/using-svg-symbols-as-css-backgrounds). Perhaps combining it with [scaled background image SVGs](https://developer.mozilla.org/en-US/docs/Web/CSS/Scaling_of_SVG_backgrounds) is a nice solution. Will have to try that.

I wasn't thinking clearly about the GZIP-ing. Thanks for the clarification. I was looking at the similar CSS handling code, which un-GZIPs the CDN-ized asset before inlining it into the document. But that's not really relevant in this case.

**Comment by mainej** [comment]
Figured out how to fetch SVG sprite from an external source... sorta. Just realized it can't use our CSS `class`es, which should have been obvious. This means moving the classes back into the CLJS, when possible, and inlining color codes otherwise.

This approach doesn't need background images, which is convenient, though I think background images can use the same technique. But, it's reported to not work in IE (though it does work in Edge). Also some reports of problems in older Android, so we'd have to test carefully. If it's a problem, we can fetch the SVG with JS, and stick it somewhere in the DOM. That approach might be the best all-around, because it would let us use the CSS classes.

File: resources/public/content/guarantee.html [hunk 1]
```diff
@@ -1,9 +1,10 @@
...
+        <svg style="width: 154px; height: 155px" class="fill-green">
+            <use xlink:href="/images/sprites.svg#guarantee"></use>
```
Comment (mainej) [inline resources/public/content/guarantee.html]: TODO: use asset url helper

**Comment by mainej** [comment]
OK, time to make a decision on merging this. Until we see it on acceptance I'm not sure we'll be able to test the Android stuff or see how it affects JS/HTML size.

I've decided not to move the spinner—the animations aren't working and it's used in views.cljs in a way that is hard to refactor.

Other SVGs not in the sprite: 
- The large spinner, though it could be moved, is nice to have live next to the main spinner
- The Mayvenn logo used while loading client-side-only pages. Again, views.cljs weirdness.
- The carousel arrows. We're using them as background images to override the CSS provided by slick.css

**Comment by sheelc** [comment]
Awesome, we'll do a chore tomorrow and you and your pair can take a shot at
merging/getting the cross browser issues sorted.

**Pull request merged**